src/Aspire.Cli/Commands/UpdateCommand.cs

```diff
@@ -233,15 +246,27 @@ private async Task<int> ExecuteSelfUpdateAsync(ParseResult parseResult, Cancella
     {
         var channel = selectedChannel ?? parseResult.GetValue<string?>("--channel") ?? parseResult.GetValue<string?>("--quality");
 
-        // If channel is not specified, prompt the user
+        // If channel is not specified, check for hives and either prompt or use default
         if (string.IsNullOrEmpty(channel))
         {
-            var channels = new[] { "stable", "staging", "daily" };
-            channel = await InteractionService.PromptForSelectionAsync(
-                "Select the channel to update to:",
-                channels,
-                q => q,
-                cancellationToken);
+            // If there are hives (PR build directories), prompt for channel selection.
+            // Otherwise, use "stable" as the default channel.
+            var hasHives = ExecutionContext.HasHives;
```

Comment:
Why is HasHives always assigned to a variable before use?

edit: I see it accesses disk. I think the property should be changed to a method